src/Security/Authentication/Provider/ContaoAuthenticationProvider.php

```diff
@@ -29,82 +32,177 @@
 class ContaoAuthenticationProvider extends DaoAuthenticationProvider
 {
     /**
-     * @var Session
+     * @var ContaoFrameworkInterface
      */
-    private $session;
+    private $framework;
 
     /**
      * @var TranslatorInterface
      */
     private $translator;
 
     /**
-     * @var string
+     * @var RequestStack
      */
-    private $providerKey;
+    private $requestStack;
 
     /**
-     * @var ContaoFrameworkInterface
+     * @var \Swift_Mailer
      */
-    private $framework;
+    private $mailer;
 
     /**
-     * @var LoggerInterface|null
+     * @var array
      */
-    private $logger;
+    private $options;
+
+    /**
+     * @var array
+     */
+    private $defaultOptions = [
+        'login_attempts' => 3,
+        'lock_period' => 60 * 5,
+    ];
 
     /**
      * @param UserProviderInterface    $userProvider
      * @param UserCheckerInterface     $userChecker
      * @param string                   $providerKey
      * @param EncoderFactoryInterface  $encoderFactory
-     * @param bool                     $hideUserNotFoundExceptions
-     * @param Session                  $session
-     * @param TranslatorInterface      $translator
      * @param ContaoFrameworkInterface $framework
-     * @param LoggerInterface|null     $logger
+     * @param TranslatorInterface      $translator
+     * @param RequestStack             $requestStack
+     * @param \Swift_Mailer            $mailer
+     * @param array                    $options
      */
-    public function __construct(UserProviderInterface $userProvider, UserCheckerInterface $userChecker, $providerKey, EncoderFactoryInterface $encoderFactory, $hideUserNotFoundExceptions, Session $session, TranslatorInterface $translator, ContaoFrameworkInterface $framework, LoggerInterface $logger = null)
+    public function __construct(UserProviderInterface $userProvider, UserCheckerInterface $userChecker, $providerKey, EncoderFactoryInterface $encoderFactory, ContaoFrameworkInterface $framework, TranslatorInterface $translator, RequestStack $requestStack, \Swift_Mailer $mailer, array $options = [])
     {
-        parent::__construct($userProvider, $userChecker, $providerKey, $encoderFactory, $hideUserNotFoundExceptions);
+        parent::__construct($userProvider, $userChecker, $providerKey, $encoderFactory, false);
 
-        $this->logger = $logger;
-        $this->session = $session;
-        $this->translator = $translator;
-        $this->providerKey = $providerKey;
         $this->framework = $framework;
+        $this->translator = $translator;
+        $this->requestStack = $requestStack;
+        $this->mailer = $mailer;
+        $this->options = array_merge($this->defaultOptions, $options);
     }
 
     /**
      * {@inheritdoc}
      */
     public function checkAuthentication(UserInterface $user, UsernamePasswordToken $token): void
     {
+        if (!$user instanceof User) {
+            parent::checkAuthentication($user, $token);
```

Comment:
Could be shortened to `return parent::checkAuthentication($user, $token);`.

Reply:
I don't think so.

Reply:
PhpStorm would generate a warning because the `checkAuthentication` method does not have a return value …